### PR TITLE
Change redirect host from main domain name to requested domain name

### PR DESCRIPTION
**Pull request by hypernics**
When you have enabled "SSL redirect" and have alias domain(s) always redirected to the main https domain: http://maindomain.com -> https://maindomain.com and http://aliasdomain.com -> https://maindomain.com

I think the required operation is to redirect to the requested domain: http://maindomain.com -> https://maindomain.com and http://aliasdomain.com -> https://aliasdomain.com

I tested with nginx, I edited an apache config file with the generated text, but I don't have lighthttpd.

### CODE REVIEW

File: scripts/jobs/cron_tasks.inc.http.20.lighttpd.php [hunk 1]
```diff
@@ -435,11 +435,13 @@ protected function getVhostContent($domain, $ssl_vhost = false, $ipid)
 
 			// Get domain's redirect code
 			$code = getDomainRedirectCode($domain['id'], '301');
-
-			$vhost_content .= '  url.redirect-code = ' . $code. "\n";
-			$vhost_content .= '  url.redirect = (' . "\n";
-			$vhost_content .= '     "^/(.*)$" => "' . $uri . '$1"' . "\n";
-			$vhost_content .= '  )' . "\n";
+            
+            $vhost_content .= '  $HTTP["host"] =~ "^(.*)$" {'. "\n";
```
Comment (d00p) [inline scripts/jobs/cron_tasks.inc.http.20.lighttpd.php]: why the "match-all"-regex? unnecessary?!

**Comment by d00p** [comment]
Looks okay so far

File: scripts/jobs/cron_tasks.inc.http.20.lighttpd.php (hunk 1 above)
Comment (d00p) [inline scripts/jobs/cron_tasks.inc.http.20.lighttpd.php]: okay, so, needed for thereplacer, understood. But if i understand the doc correctly, shouldn't the replacer be 0% ? dont know anyone using lighttpd :p

Comment (hypernics) [inline scripts/jobs/cron_tasks.inc.http.20.lighttpd.php]: This code use %1/$1 with brackets in match string: http://superuser.com/questions/317075/redirect-from-http-to-https-in-lighthttpd/319357#319357
:confused:

**Comment by hypernics** [comment]
Tested with lighttpd too.

**Comment by hypernics** [comment]
ready to merge if you want, tested with all 3 webserver

**Pull request merged**